package/base-files/files/etc/init.d/sysctl

```diff
@@ -36,9 +37,50 @@ apply_defaults() {
 	fi
 }
 
-start() {
+handle_sysctl() {
+	local cfg="$1"
+	local sysctl="$2"
+
+	local enable section value
+
+	config_get_bool enable "$cfg" enable '0'
+	config_get section "$cfg" section
+
+	if [ "$enable" = '0' ]; then
+		# get default value
+		value="$(grep "$section" /tmp/sysctl-default.conf)"
+		value="${value##*= }"
+	else
+		config_get value "$cfg" value
+	fi
+
+	[ -z "$section" ] || [ -z "$value" ] && return
+
+	echo "${section}=${value}" >> "$sysctl"
+}
+
+apply_uci() {
+	mkdir -p /tmp/sysctl.d
+
+	rm -f /tmp/sysctl.d/01-uci.conf
+
+	config_load sysctl
+	config_foreach handle_sysctl sysctl "/tmp/sysctl.d/01-uci.conf"
```

Comment:
Why do we need all this complexity with tmp? Can't we just apply the value directly after the sysctl ones are applied?

Reply:
If someone deletes the configuration, then I thought the default/startup value should be set again as before the change.

Otherwise I have to reboot every time I delete the value in the uci configuration so that I have the default value of Kernel again.

Reply:
ugh sad that they never implemented a way in sysctl to reset a value... I still thing the logic can be reduced to just the line you added

`[ -f /tmp/sysctl-default.conf ] || sysctl -a 1>/tmp/sysctl-default.conf 2>/dev/null`

and change it to something like

```
if [ -f /tmp/sysctl-default.conf ]; then
  sysctl -p /tmp/sysctl-default.conf
else
  sysctl -a 1>/tmp/sysctl-default.conf 2>/dev/null
fi
```

We store in sysctl-default.conf the default values and we just repply them before any modification from packages and uci. And with this change, we can just parse the uci option and apply them directly.

Also I think we should cache the default sysctl after apply_defaults.

Or even move the apply_defaults and the sysctl-default.conf creation in start_service. And make reload_service only apply the file.